### PR TITLE
feat(pyats): add architecture-agnostic defaults resolution to NACTestBase

### PR DESCRIPTION
**Pull request by aitestino**
## Description

This PR adds architecture-agnostic defaults resolution utilities to `NACTestBase`, making it available to all NAC architectures (ACI, SD-WAN, Catalyst Center) through opt-in configuration.

Previously, defaults resolution was implemented in nac-test-pyats-common as ACI-specific code. This refactor moves the core resolution logic to the base framework, enabling all architectures to leverage defaults files while maintaining architecture-specific prefixes and error messages.

## Closes

- N/A

## Related Issue(s)

- Extracted from https://github.com/netascode/nac-test-pyats-common/pull/25 (closed in favor of this approach)
- Complements #525 (CLI validation for defaults files)

## Type of Change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring / Technical debt (internal improvements with no user-facing changes)
- [ ] Documentation update
- [ ] Chore (build process, CI, tooling, dependencies)
- [ ] Other (please describe):

## Test Framework Affected

- [x] PyATS
- [ ] Robot Framework
- [ ] Both
- [ ] N/A (not test-framework specific)

## Network as Code (NaC) Architecture Affected

- [ ] ACI (APIC)
- [ ] NDO (Nexus Dashboard Orchestrator)
- [ ] NDFC / VXLAN-EVPN (Nexus Dashboard Fabric Controller)
- [ ] Catalyst SD-WAN (SDWAN Manager / vManage)
- [ ] Catalyst Center (DNA Center)
- [ ] ISE (Identity Services Engine)
- [ ] FMC (Firepower Management Center)
- [ ] Meraki (Cloud-managed)
- [ ] NX-OS (Nexus Direct-to-Device)
- [ ] IOS-XE (Direct-to-Device)
- [ ] IOS-XR (Direct-to-Device)
- [ ] Hyperfabric
- [x] All architectures
- [ ] N/A (architecture-agnostic)

## Platform Tested

> nac-test supports macOS and Linux only

- [x] macOS (version tested: )
- [x] Linux (distro/version tested: Ubuntu 24.04)

## Key Changes

- **New module**: `nac_test/pyats_core/common/defaults_resolver.py`
  - Pure utility functions with no PyATS dependencies
  - `ensure_defaults_block_exists()`: Validates defaults block presence
  - `get_default_value()`: Single-path and cascade lookup with JMESPath

- **NACTestBase enhancements**: `nac_test/pyats_core/common/base_test.py`
  - New class attributes: `DEFAULTS_PREFIX`, `DEFAULTS_MISSING_ERROR`
  - New instance method: `get_default_value(*paths, required=True)`
  - Opt-in architecture: raises `NotImplementedError` if `DEFAULTS_PREFIX` is `None`

- **Comprehensive test coverage**: 63 tests (51 unit + 12 integration)
  - Tests for pure utility functions (architecture-agnostic)
  - Tests for NACTestBase wrapper integration
  - Edge cases: falsy values, deep nesting, cascade behavior

## Testing Done

- [x] Unit tests added/updated
- [x] Integration tests performed
- [ ] Manual testing performed:
  - [ ] PyATS tests executed successfully
  - [ ] Robot Framework tests executed successfully
  - [ ] D2D/SSH tests executed successfully (if applicable)
  - [ ] HTML reports generated correctly
- [x] All existing tests pass (`pytest` / `pre-commit run -a`)

### Test Commands Used

```bash
# Run all new tests
uv run pytest tests/unit/pyats_core/common/ -v

# Results: 63 passed in 0.14s
# - 51 unit tests for defaults_resolver module
# - 12 integration tests for NACTestBase wrapper
```

## Checklist

- [x] Code follows project style guidelines (`pre-commit run -a` passes)
- [x] Self-review of code completed
- [x] Code is commented where necessary (especially complex logic)
- [ ] Documentation updated (if applicable)
- [x] No new warnings introduced
- [ ] Changes work on both macOS and Linux
- [ ] CHANGELOG.md updated (if applicable)

## Screenshots (if applicable)

N/A - This is infrastructure code with no UI changes.

### Architecture Design

This implementation follows the **opt-in pattern** used elsewhere in nac-test:
- Base class provides functionality but requires subclass configuration
- Architectures enable by setting `DEFAULTS_PREFIX` class attribute
- Raises `NotImplementedError` if feature is used without configuration

### Migration Path for nac-test-pyats-common

After this PR merges, nac-test-pyats-common will:
1. Remove `nac_test_pyats_common/aci/defaults_resolver.py`
2. Update `APICTestBase` to set `DEFAULTS_PREFIX = "defaults.apic"`
3. Inherit `get_default_value()` from `NACTestBase`

### Cascade/Fallback Support

Supports multiple JMESPaths with first-found-wins semantics:
```python
# Try specific path first, fall back to general path
default_pod = self.get_default_value(
    "tenants.l3outs.nodes.pod",
    "tenants.nodes.pod"
)
```

### Type Safety

Return type is `Any | None` because JMESPath can return any type (str, int, bool, dict, list). This is intentional - the type depends on the data model structure at the queried path.

### CODE REVIEW

File: nac_test/pyats_core/common/base_test.py
Comment (oboehmer) [inline nac_test/pyats_core/common/base_test.py]: I know this is not related to this PR, but when looking at this: detecting the same controller type by every tests sounds redundant.. I know the performance impact here is negligible, but maybe we can start to inject shared run-time "constants" via the pyats context once and re-use it there?

File: tests/unit/pyats_core/common/test_base_test_iosxe_credentials.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        nac_test_base_class: Any,
+        temp_data_model_file: str,
+        iosxe_controller_env: None,
+    ) -> None:
+        """setup() should also work if IOSXE USERNAME/PASSWORD are provided.
+
+        While not required, if someone sets them, we should accept them.
+        """
+        # Verify all credentials are set
+        assert "IOSXE_URL" in os.environ
```
Comment (oboehmer) [inline tests/unit/pyats_core/common/test_base_test_iosxe_credentials.py]: side note: I recall this was changed to `IOSXE_HOST` with the introduction of netconf as an alternative connection method in XE-as-code?

File: tests/unit/pyats_core/common/test_defaults_resolver.py [hunk 2]
```diff
@@ -0,0 +1,854 @@
...
+"""Unit tests for defaults_resolver module.
+
+This module tests the pure utility functions for reading default values
+from merged NAC data models. These functions use JMESPath for data model
+traversal and are architecture-agnostic.
+
+Test Structure:
+    - TestEnsureDefaultsBlockExists: Tests for defaults block validation
+    - TestGetDefaultValueSinglePath: Tests for single-path lookups
+    - TestGetDefaultValueCascade: Tests for cascade/fallback behavior
```
Comment (oboehmer) [inline tests/unit/pyats_core/common/test_defaults_resolver.py]: I feel both these test classes could leverage `pytest.mark.parametrize` as they have plenty of repeated test patterns
